azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -264,14 +264,20 @@ def on_message(client, userdata, mqtt_message):
         mqtt_client.on_publish = on_publish
         mqtt_client.on_message = on_message
 
+        # Set paho automatic-reconnect delay to 2 hours.  Ideally we would turn
+        # paho auto-reconnect off entirely, but this is the best we can do.  Without
+        # this, we run the risk of our auto-reconnect code and the paho auto-reconnect
+        # code conflicting with each other.
+        mqtt_client.reconnect_delay_set(120 * 60)
+
         logger.debug("Created MQTT protocol client, assigned callbacks")
         return mqtt_client
 
-    def _stop_automatic_reconnect(self):
+    def _cleanup_transport_on_error(self):
         """
```

Comment:
Why the name change?

Reply:
Usage of this function has expanded.  It no longer exists solely to stop the reconnect.  Most of that is handled by setting the interval to 2 hours.  Now this function is used to reset paho to a clean state (as much as we can, anyway).

---
In reply to: [399534294](https://github.com/Azure/azure-iot-sdk-python/pull/516#discussion_r399534294) [](ancestors = 399534294)